torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -1979,7 +1954,24 @@ void ONNXShapeTypeInference(
     for (auto output : clone_node->outputs()) {
       n_graph->registerOutput(output);
     }
-
+    for (size_t i = 0; i < clone_node->outputs().size(); ++i) {
+      torch_to_onnx_output_name[n->output(i)->debugName()] = clone_node->output(i)->debugName();
+    }
+    // Make inferred_shape_data use name from ONNX graph instead of original PyTorch graph
+    // Use original_keys for removing original data which is duplicate
+    std::vector<string> original_keys;
+    for (const auto& gs_data: inferred_shape_data) {
+      const auto onnx_output_name = torch_to_onnx_output_name.find(gs_data.first);
+      if (onnx_output_name != torch_to_onnx_output_name.end()) {
+        inferred_shape_data[onnx_output_name->second] = gs_data.second;
```

Comment:
is it possible to have name collision?

Reply:
actually I'm a little confused. all `onnx_output_name` are from `torch_to_onnx_output_name`, which is populated locally from `n->outputs()`. This would imply the loop here is only mapping names for `n->outputs()`. Hence two questions
1. Why would shape data already exist for `n->outputs()` before doing shape inference?
2. If answer to above is yes, why is it necessary to do this for `n->outputs()`? And why aren't `n->inputs()` considered?

Reply:
1. Yes it's possible to have name collision.
2. Thank you so much for catching this! I did some debugging and yes we should do it for `n->inputs()` instead of `n->outputs()` because the current input comes from previous propagated output. It was just accidentally worked somehow. Therefore, I have used input map to change previous inferred shapes instead of output map.